staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances.go

```diff
@@ -281,25 +281,26 @@ func (az *Cloud) InstanceID(ctx context.Context, name types.NodeName) (string, e
 			return "", fmt.Errorf("no credentials provided for Azure cloud provider")
 		}
 
-		// Get resource group name.
+		// Get resource group name and subscription ID.
 		resourceGroup := strings.ToLower(metadata.Compute.ResourceGroup)
+		subscriptionID := strings.ToLower(metadata.Compute.SubscriptionID)
 
 		// Compose instanceID based on nodeName for standard instance.
-		if az.VMType == vmTypeStandard {
-			return az.getStandardMachineID(resourceGroup, nodeName), nil
+		if metadata.Compute.VMScaleSetName == "" {
```

Comment:
why not use `if az.VMType == vmTypeStandard`?

Reply:
this increase robustness, so that even vmtype is not configured, we could still figure out the right vmtype.

Reply:
I did run into the issue with vmtype not being specified for scalesets, so great addition